waltz-common/src/main/java/com/khartec/waltz/common/StringUtilities.java

```diff
@@ -113,4 +118,19 @@ public static int length(String str) {
                 ? 0
                 : str.length();
     }
+
+
+    public static String join(Collection<?> values, String separator) {
+        return values.stream()
+                .filter(v -> v != null)
+                .map(v -> v.toString())
+                .collect(Collectors.joining(separator));
+    }
+
+
+    public static <T> List<T> split(String value, String separator, Function<String, T> fromString) {
```

Comment:
not bothered about adding `split` function - it's the split with a sneaky `map` built in !  Perhaps simply a more explicit name so the signature is less mysterious.   

`splitThenMap(String str, String separator, Function<String, T> itemTransformer)`